my_each.rb

```diff
@@ -1,6 +1,143 @@
 module Enumerable
-    def my_each(array)
-      for i in 0...array.length
-        p "gaa"
+  def my_each
+    (0...length).each do |i|
+      yield self[i]
     end
-end
\ No newline at end of file
+    self
+  end
+
+  def my_each_with_index
+    (0...length).each do |i|
+      yield self[i], i
+    end
+    self
+  end
+
+  def my_select
+    new_array = []
+    my_each do |element|
+      new_array.push(element) if yield element
+    end
+    new_array
+  end
+
+  def my_all?
+    result = true
+    my_each do |element|
+      statement = if block_given?
+                    yield element
+                  else
+                    element
+                  end
+
+      unless statement
+        result = false
+        break
+      end
+
+      result = true
+    end
+    result
+  end
+
+  def my_any?
+    result = true
+    my_each do |element|
+      statement = if block_given?
+                    yield element
+                  else
+                    element
+                  end
+
+      if statement
+        result = true
+        break
+      end
+      result = false
+    end
+    result
+  end
+
+  def my_none?
+    !my_any? do |element|
+      yield element
+    end
+  end
+
+  def my_count
+    counter = 0
+    my_each do |element|
+      param = yield element
+      counter += 1 if param.eql?(true) || param.eql?(element)
+    end
+    counter
+  end
+
+  def my_map(proc = nil)
+    new_arr = []
+    my_each do |element|
+      param = if proc
+                proc.call(element)
+              else
+                yield element
+              end
+      new_arr.push(param)
+    end
+    new_arr
+  end
```

Comment:
- [ ] should have working `my_map` **(covered by automated tests)** identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method should return an enumerator if no block or argument is given.